pkg/services/dashboards/service/client/client.go

```diff
@@ -26,8 +27,22 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+const (
+	// maxConcurrentGetRequests is the maximum number of concurrent Get requests to the K8s API.
+	// It is used to determine max amount of concurrent version fallback requests when handling a List request.
+	// TODO: do we need to make this configurable via config.ini?
+	maxConcurrentGetRequests = 10
```

Comment:
Yeah I think it makes sense to have this configurable - but it can be done in a followup PR